api/v1/images.rb

```diff
@@ -113,6 +113,49 @@ class MissingUploadedFileError < Exception; end
     end
   end
 
+
+  # @apidoc
+  # Delete a single image from S3.
+  #
+  # @category Tiramisu
+  # @path /api/tiramisu/v1/images/:path
+  # @http DELETE
+  #
+  # @required [String] path The whole image URL minus hostname, e.g. apdm/oa/kittens/20160208121232-809-a5x0/original.jpg.
+  # @example /api/tiramisu/v1/images
+  # @status 200 [JSON] {deleted: true} if the delete operation succeeded, else an error message.
+  delete '/images' do
+    content_type 'application/json', :charset => 'utf-8'
+
+    path = params[:path]
+    halt [400, {error: 'no path'}.to_json] unless path
+    halt [403, {error: 'You must be god to delete anything'}.to_json] unless identity_is_god?
+
+    LOGGER.info "Delete image at #{path}"
+    # TODO: halt 403 unless god
+
+    begin
+      # Delete file from Amazon S3.
+      delete_ok = asset_store.delete path
+      [200, {deleted: delete_ok}.to_json]
+
+    rescue S3::Error::NoSuchKey => e
+      message = "Unknown image: #{path}"
+      LOGGER.warn message
+      [404, {error: message}.to_json]
+
+    rescue S3::Error::AccessDenied => e
+      message = "#{e.message} at S3 for path: #{path}. Key/bucket trouble?"
+      LOGGER.warn message
+      [403, {error: message}.to_json]
+
+    rescue => e
+      LOGGER.warn e.message
+      LOGGER.respond_to?(:exception) ? LOGGER.exception(e) : LOGGER.error(e)
```

Comment:
Won't this return 200 (or invoke some other default behavior?) since you're not halting? I believe Sinatra apps catch errors and send them to `LOGGER.exception` anyway.